### DISCUSSION TITLE
Request node JSON body using variables or payload

### DISCUSSION THREAD

**Discussion by fuzzydunlop1976** [alexkn/node-red-contrib-homeconnect#87] [Q&A]
Hi,

I've just started to play around with node-red and this contrib is helping me learn - many thanks for all of the work that must have gone into it. I'm in awe of the possibilities that open up. One quick question.

I've managed to start my washer using the 'request' node, selecting the 'program' tag, using the 'start_program' action, giving my HAID and then providing the program name formatted as 'data' and 'key' that I want to start in the JSON body. 

Is it possible to reference either the incoming msg payload or a variable in the body or does it need to be written as JSON each time?

Earlier in my flow, I'd saved the program that had been set on the machine when the door closes as a flow variable object. I was hoping to be able to utilise this object back in the request node when the time comes to actually start the washer. The error messages suggest the field is expecting JSON - Could anyone point me in the direction of if this is possible?

**Comment by Cafun** [comment]
Unfortunately I cannot help you (atm) with your question, but maybe if you can help me with my own (see my [discussion](https://github.com/alexkn/node-red-contrib-homeconnect/discussions/83). Can you send me an export from your flow using program tag etc?)
Maybe then I can see what you mean.

**Reply by fuzzydunlop1976** [reply]
I'd used exactly the same format as you to start a program:

'Program' selected within the tag field. 
'Start_Program' within the action field and then 
a JSON 'data' object containing 'key' and the "LaudryCare.Washer.Program.XXXXX" as the value. 

I didn't use any 'options' although I was intending to try this at some point.

I was trying to avoid hard coding the JSON body for each program I wanted to run. Instead, my idea was to save the 'selected program state' as a global or flow variable when the washing machine door was closed. I would then like to use this variable and pass into the start_program action when another trigger occurs (e.g leave the house).

I think the fields in the home-connect nodes though seem to expect 'plain' JSON, so I'm not sure that what I want to do is possible.

**Reply by jslits** [reply]
any luck? I'm looking for the same: how to add msg values to the body

**Reply by Cafun** [reply]
No luck yet unfortunately. I didn't had time to looking in to it any further.
